### PR TITLE
rename preserved logins to fireproofing and use protocol where possible

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/392891325557410/1208796954308702/f
Tech Design URL:
CC: @bwaresiak 

**Description**:
Rename preserved logins to fireproofing and use protocol where possible

This is stacked on https://github.com/duckduckgo/iOS/pull/3590 - am trying to make baby steps towards removing a few of the gnarlier singletons (Favicons and Fireproofing at least)

**Steps to test this PR**:
1. Smoke test fireproofing with cookies, you can use https://www.setcookie.net (remember to set an expiry date for it to work)
2. Go into Settings > Data Clearing > Fireproof sites and make sure fire proofed sites are visible and can be deleted from there
3. Check it works and visible in the settings UI

### CODE REVIEW

File: DuckDuckGo/FireproofingSettingsViewController.swift [hunk 1]
```diff
@@ -35,7 +35,10 @@ class PreserveLoginsSettingsViewController: UITableViewController {
 
     var model = [String]()
     private var shouldShowRemoveAll = false
-    
+
+    // TODO inject this
```
Comment (jaceklyp) [inline DuckDuckGo/FireproofingSettingsViewController.swift]: inject this ;)

File: DuckDuckGo/TabViewController.swift [hunk 2]
```diff
@@ -1621,7 +1624,7 @@ extension TabViewController: WKNavigationDelegate {
     }
     
     private func checkLoginDetectionAfterNavigation() {
-        if preserveLoginsWorker?.handleLoginDetection(detectedURL: detectedLoginURL,
+        if fireproofingWorker?.handleLoginDetection(detectedURL: detectedLoginURL,
                                                       currentURL: url,
                                                       isAutofillEnabled: AutofillSettingStatus.isAutofillEnabledInSettings,
                                                       saveLoginPromptLastDismissed: saveLoginPromptLastDismissed,
```
Comment (jaceklyp) [inline DuckDuckGo/TabViewController.swift]: nitpick; some indentation issue

File: DuckDuckGoTests/WebCacheManagerTests.swift [hunk 3]
```diff
@@ -205,18 +206,36 @@ class WebCacheManagerTests: XCTestCase {
             
     // MARK: Mocks
     
-    class MockPreservedLogins: PreserveLogins {
+    class MockFireproofing: Fireproofing {
```
Comment (jaceklyp) [inline DuckDuckGoTests/WebCacheManagerTests.swift]: from the changes it looks like you're no longer using it

File: DuckDuckGoTests/WebCacheManagerTests.swift [hunk 4]
```diff
@@ -62,9 +63,9 @@ class WebCacheManagerTests: XCTestCase {
     @available(iOS 17, *)
     @MainActor
     func testWhenCookiesHaveSubDomainsOnSubDomainsAndWidlcardsThenOnlyMatchingCookiesRetained() async throws {
-        let logins = MockPreservedLogins(domains: [
-            "mobile.twitter.com"
-        ])
+        let fireproofing = UserDefaultsFireproofing.shared
+        fireproofing.clearAll()
+        fireproofing.addToAllowed(domain: "mobile.twitter.com")
```
Comment (jaceklyp) [inline DuckDuckGoTests/WebCacheManagerTests.swift]: im not sure i like it, if we ever decide to run our tests concurrently (to speed them up) using a shared store doesn't sound like a good option

**Review by jaceklyp — CHANGES_REQUESTED** [review]

File: DuckDuckGoTests/WebCacheManagerTests.swift (hunk 4 above)
Comment (brindy) [reply DuckDuckGoTests/WebCacheManagerTests.swift]: The problem is the previous mock was a subclass of the actual thing so these tests are a bit 💩  anyway.  Let me see if I can mock it properly.

Comment (brindy) [reply DuckDuckGoTests/WebCacheManagerTests.swift]: I decided to revert to the subclass mock, I don't want to change *too* much with this PR.

**Review by jaceklyp — APPROVED** [review]
LGTM :)

**Pull request merged**